### DISCUSSION TITLE
Download Failed: Error when loading first segment, Error opening input: Invalid data found when processing input

### DISCUSSION THREAD

**Discussion by ashishdhiman2007** [aclap-dev/video-downloadhelper#611] [Help and Q&A]
### Your question

I tried downloading video course for my personal storage but it gives me error. Please guide and help. If you need more data please ask. Thanks

{
  "version": "9.0.1.18",
  "target": "google",
  "channel": "stable",
  "page_url": "https://courses.acharyaganesh.com/s/courses/658578c2e4b077109bc75e43/take",
  "error": {
    "details": "SideDownload error: : [hls @ 0x7f9de7b048c0] Error when loading first segment 'https://dnqe0v482iyb1.cloudfront.net/spees/w/o/63280acce4b0d6384cb61e0b/v/65ae22c5e4b09407e2744497/u/64f15a1ae4b0af98c8e80f75/t/748cd0b5dad28787c2bfd04d2a04e956/p/assets/videos/63280acce4b0d6384cb61e0b/2024/01/22/65ae22c5e4b09407e2744497/hls_1M_000.ts'\n[in#0 @ 0x7f9de7b04780] Error opening input: Invalid data found when processing input\nError opening input file https://dnqe0v482iyb1.cloudfront.net/spees/w/o/63280acce4b0d6384cb61e0b/v/65ae22c5e4b09407e2744497/u/64f15a1ae4b0af98c8e80f75/t/748cd0b5dad28787c2bfd04d2a04e956/p/assets/videos/63280acce4b0d6384cb61e0b/2024/01/22/65ae22c5e4b09407e2744497/hls_1M_.m3u8.\nError opening input files: Invalid data found when processing input\n",
    "downloadable_id": "downloadable_309171792225506",
    "error": "coapp_failure",
    "id": "coapp_failure_2829bdd5-764c-40a7-811c-850017934764"
  }
}

### Addon & CoApp version and details

version: 9.0.1.18
target: google
channel: stable
lang: en-GB
coapp: {"found":true,"new_version":false,"path":"/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp","version":"2.0.19"}
license: {"unset":true}
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36
{
    "used_sidebar_button": true,
    "download_directory": "/Users/netsmartz/dwhelper",
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": true,
    "open_count_store": 6
}

**Comment by Mattmjs** [comment]
Another "unable to open key file" according to the error message.

#EXT-X-KEY:METHOD=AES-128,URI="k/timestamp",IV=0x496daa1c6914000e408c65cead91fc29

**Reply by ashishdhiman2007** [reply]
Is this a common scenario, which multiple users are facing? Is there a way out?

**Reply by Mattmjs** [reply]
It seems on some course platforms and it does not look like there is a solution.

**Reply by GrampaWildWilly** [reply]
I tried to use ffmpeg directly on the HLS manifest in the error message in the original post.  This happens to be a stream manifest, not a master manifest.  The error generated trying to read the key file was 502 Bad gateway.  The site where the key file apparently resides is Cloudfront, one of the big cloud services.  I don't know how a cloud service could have a bad gateway.  It seems like an intentional error to me, intended expressly to thwart downloading.  On the other hand, I haven't registered for whatever course this is.  It's possible, although I can't even estimate at what odds, that you could get in there while you're logged in.  You might need to explicitly code your user ID & password in the URL.  That would be in this form:

https://yourID:yourpassword@theURLofthemanifest

If either your user ID or your password or both contain @, you have to replace those with %40.  The @ is the delimiter between the password & the URL, so you would confuse things by including any @ signs in your user ID & password, for example, if your user ID is your E-mail address.

If you are so motivated, you can read my tutorial on ffmpeg.  Do a Google search with "vdh ffmpeg" as the search key & the first search result will be my tutorial.  It depends on how important having a copy of this course is to you.  But I make no guarantees.  The course web site may have measures in place to prevent downloading, measures that might include DRM & might not.  But that's all speculation subject to proof that only you can provide, as a registered user of the site.  Only you can determine for certain whether ffmpeg can succeed on this site where VDH fails.
